### ISSUE TITLE
Named semaphore generation on Mac OS X

### ISSUE THREAD

**Issue by alemenke** [MultiChain/multichain#33]
The generation of semaphore names for OS X is making use of ```rand()``` without seeding it, which ultimately leads to multichain always trying to use the same semaphore name on each invocation until the machine is rebooted.  https://github.com/MultiChain/multichain/blob/0a428c7f74e600e242a519e21bf1123cdd76a94c/src/utils/systemdependent.cpp#L119-L143

This becomes a problem when the semaphore isn't cleaned up properly (e.g. after a SIGKILL or even when stopping a running sync with SIGINT, deleting the datadir and trying to sync again, which can get pretty annoying when testing stuff) as multichaind will refuse to start until the machine is rebooted or the semaphore is removed manually

I guess it would be a good idea to add ```mc_RandomSeed(mc_TimeNowAsUInt())``` in init to prevent issues like that

**Comment by gidgreen** [comment]
Noted - thanks for the comment.

**Comment by bitcartel** [comment]
@gidgreen I agree the change should be made, adding `mc_RandomSeed(mc_TimeNowAsUInt())` in init would be best, so that all code paths and not just semaphore creation will benefit.

**Comment by bitcartel** [comment]
@alemenke What version of macOS are you experiencing this issue with, and are you running macOS inside a VM?  Also, `mc_RandomSeed` is already being invoked here: https://github.com/MultiChain/multichain/blob/master/src/chainparams/params.cpp

**Comment by alemenke** [comment]
> @alemenke What version of macOS are you experiencing this issue with, and are you running macOS inside a VM?
OS X Sierra 10.12.6. Happens both on a Mac Mini and a CI VM I've been testing with

> Also, mc_RandomSeed is already being invoked here: https://github.com/MultiChain/multichain/blob/master/src/chainparams/params.cpp#L274
Yes, but it does not seem like it is invoked when trying to start a fresh sync/connecting to an existing network without a datadir. It is invoked when starting multichaind with a pre-existing datadir, but I haven't tested if it is related to the network config file or the permissions.db (-reindex would probably show that).
I've added debug output to both `mc_RandomSeed` and `__US_SemCreate` when I initially looked for the problem and have verified that it is not called and that the `rand` function has returned the same indices in the semaphore name generation code
